paket.dependencies

```diff
@@ -1,35 +1,59 @@
 source http://nuget.org/api/v2
+redirects: on
+strategy: max
 
-nuget FSharp.Compiler.Service
 nuget Fantomas
+nuget FSharp.Compiler.Service
+nuget FSharp.Core
+nuget FSharp.Management
 nuget FSharp.ViewModule.Core
-nuget FsXaml.Wpf
-nuget SourceLink.SymbolStore 0.5.0
 nuget FSharpLint.Core prerelease
-nuget VSSDK.IDE.Deploy.14
-nuget FSharp.Management
 nuget FsPickler
-nuget FSharp.Core redirects:on
+nuget FsXaml.Wpf
+nuget SourceLink.SymbolStore 0.5.0
 
+group vssdklegacy
+    source http://nuget.org/api/v2
+    framework >= net40
+    
+    nuget Microsoft.VisualStudio.Shell.Interop.8.0
+    nuget Microsoft.VisualStudio.Shell.Immutable.10.0
+    nuget VSSDK.ComponentModelHost.11
+    nuget VSSDK.DTE
+    nuget VSSDK.DTE.8
+    nuget VSSDK.DTE.10
+    nuget VSSDK.NavigateTo
+    nuget VSSDK.VSLangProj
+    
+    
+group vssdk
+    source http://nuget.org/api/v2
+    framework >= net45
+    
+    nuget Microsoft.VisualStudio.Editor
+    nuget Microsoft.VisualStudio.Language.Intellisense
```

Comment:
@smoothdeveloper 

Again sorry about my Paket ignorance here.  But doesn't the lock file represent the result of a restore vs. specifying what packages are needed?  If so it seems that if a new version of these packages are uploaded then a restore could reference them and only change the lock file.  Is that correct?  

Basically I'm interested in what mechanism would stop the packages from silently upgrading to 15.0.  

> I think with the qualified paths (not based on assembly name resolution but actual path to the file) to the assemblies in *proj files it should pick up the correct versions right?

I'm not sure which value will take precedence in MSBuild: `HintPath` or values returned from the SDK.  I believe `HintPath` does take precedence today but I wouldn't bet on it, definitely wouldn't bet on it being true forever.

This is part of the reason I eventually switched to always using strongly named references for the VS editor binaries.  It allowed me to take MSBuild resolution out of the equation.  No matter what the rules are it will only succeed if it matches the exact assembly that I want :smile: